src/test/kotlin/racing/DistanceServiceTest.kt

```diff
@@ -0,0 +1,26 @@
+package racing
+
+import io.mockk.every
+import io.mockk.mockk
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+internal class DistanceServiceTest {
+
+    private val random = mockk<RandomImpl>()
+    private val distanceService = DistanceService(random)
+
+
+    @Test
+    fun `기준 4 보다 작은 경우 0 반환`() {
+        every { random.getNumber() } returns 3
+        assertThat(distanceService.get()).isEqualTo(0)
```

Comment:
Mock 객체를 사용하지 않고 이렇게 구현해볼 수 있습니다 :)
```suggestion
        // given
        val distanceService = DistanceService(object : Random {
            override fun getNumber(): Int {
                return 3
            }
        })

        // when
        val actualDistance = distanceService.get()

        // then
        assertThat(actualDistance).isEqualTo(0)
```

Reply:
추가로, 코틀린에서 interface에 메서드가 하나만 있는 Java 파일의 경우 SAM 기능으로 바로 람다를 넣을 수 있는데요, Kotlin interface에서는 SAM이 동작하지 않아 람다를 넣을 수 없습니다.
이런 경우 interface에 fun 키워드를 붙여 주신다면, kotlin 에서도 바로 람다를 사용할 수 있습니다.

```kotlin
fun interface Random { 
   ...
}
```

```kotlin
val distanceService = DistanceService{ 3 }
```